scanpipe/pipelines/populate_purldb.py

```diff
@@ -64,37 +62,11 @@ def populate_purldb_with_detected_purls(self):
         # package/dependency instances)
         if no_packages_and_no_dependencies:
             packages = scancode.get_packages_with_purl_from_resources(self.project)
-            self.feed_purldb(
-                packages=list(packages),
-                package_type="DiscoveredPackage",
-            )
-
-    def feed_purldb(self, packages, package_type):
-        """Feed PurlDB with list of PURLs for indexing."""
-        if not purldb.is_available():
-            raise Exception("PurlDB is not available.")
-
-        package_urls = list(set([package.purl for package in packages]))
-        self.log(f"Populating PurlDB with {len(package_urls):,d} {package_type}")
-
-        response = purldb.submit_purls(purls=package_urls)
-        queued_packages_count = response.get("queued_packages_count", 0)
-        unqueued_packages_count = response.get("unqueued_packages_count", 0)
-        unsupported_packages_count = response.get("unsupported_packages_count", 0)
-
-        if queued_packages_count > 0:
-            self.log(
-                f"Successfully queued {queued_packages_count:,d} "
-                f"PURLs for indexing in PurlDB"
-            )
-
-        if unqueued_packages_count > 0:
-            self.log(
-                f"{unqueued_packages_count:,d} PURLs were already "
-                f"present in PurlDB index queue"
-            )
+            purls = [{"purl": package.purl} for package in packages]
 
-        if unsupported_packages_count > 0:
-            self.log(
-                f"Couldn't index {unsupported_packages_count:,d} unsupported PURLs"
+            purldb.feed_purldb(
+                packages=purls,
+                chunk_size=100,
+                msg=(f"Populating PurlDB with {len(purls):,d} " "detected PURLs"),
```

Comment:
Seems like a leftover, meaning also that this section of the code is not tested ;)

Reply:
Fixed.
I'll sync with @AyanSinhaMahapatra to figure out how to test `populate_purldb_with_detected_purls` .

Reply:
@tdruez Fixed, we're now testing the `populate_purldb_with_detected_purls` with un-assembled packages.